### PR TITLE
chore(airflow): relax `semver` requirement bound

### PR DESCRIPTION
**Pull request by sbrugman**
Closes: https://github.com/kedro-org/kedro-plugins/issues/269

## Checklist

- [X] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [X] Updated the documentation to reflect the code changes
- [X] Added a description of this change in the relevant `RELEASE.md` file
- [X] Added tests to cover my changes

### CODE REVIEW

File: kedro-airflow/pyproject.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ license = {text = "Apache Software License (Apache 2.0)"}
 dependencies = [
     "kedro>=0.17.5",
     "python-slugify>=4.0",
-    "semver~=2.10",  # Needs to be at least 2.10.0 to get VersionInfo.match
+    "semver~=3.0",
```
Comment (deepyaman) [inline kedro-airflow/pyproject.toml:16]:
```suggestion
    "semver>=2.10",  # Needs to be at least 2.10.0 to make use of `VersionInfo.match`.
```
Can we just relax it instead? Feels unnecessary to drop 2.x compatibility, for anybody who may need that.

Comment (deepyaman) [reply kedro-airflow/pyproject.toml:16]: Exact text copied from https://github.com/kedro-org/kedro-viz/pull/1452/files

**Review by deepyaman — APPROVED** [review]
LGTM, thanks yet again!

**Review by astrojuanlu — APPROVED** [review]
LGTM! Good to see that `semver` 3.0 didn't introduce any breaking changes (very meta 🤓)

**Pull request merged**